### DISCUSSION TITLE
Query for one product using `producer_granule_id`?

### DISCUSSION THREAD

**Discussion by scottstanie** [earthaccess-dev/earthaccess#302] [Q&A] [answered]
Hi, is there any way of getting a query result using the  `collection_concept_id`? The equivalent API url would be 

https://cmr.earthdata.nasa.gov/search/granules.json?producer_granule_id=S1A_WV_SLC__1SSV_20190201T001401_20190201T002532_025728_02DC1A_8582&collection_concept_id=C1214470488-ASF

but I can't figure out how to pass these through the `earthaccess.DataGranules` class

**Comment by betolink** [comment]
Hi @scottstanie, sorry for the late reply, after looking at the supported methods in the class I don't think that one is there but we can include it for the next release, are there any other CMR parameters that would be useful to you or other people? 

We have `readable_granule_name` but not `producer_granule_id`, would this be another useful parameter?

**Reply by scottstanie** [reply]
Thanks for pointing that out! I hadn't seen that search term
```python
In [12]: earthaccess.search_data(granule_name='S1A_WV_SLC__1SSV_20190201T001401_20190201T002532_025728_02DC1A_8582', concept_id='C1214470488-ASF')
Granules found: 1
Out[12]: 
[Collection: {'ShortName': 'SENTINEL-1A_SLC', 'Version': '1'}
 Spatial coverage: {'HorizontalSpatialDomain': {'Geometry': {'GPolygons': [{'Boundary': {'Points': [{'Longitude': -81.154861, 'Latitude': -46.417355}, {'Longitude': -81.079514, 'Latitude': -46.592674}, {'Longitude': -80.811981, 'Latitude': -46.535709}, {'Longitude': -80.888298, 'Latitude': -46.360573}, {'Longitude': -81.154861, 'Latitude': -46.417355}]}}]}}}
 Temporal coverage: {'RangeDateTime': {'BeginningDateTime': '2019-02-01T00:14:01.000Z', 'EndingDateTime': '2019-02-01T00:25:32.000Z'}}
 Size(MB): 3459.2264013290405
 Data: ['https://datapool.asf.alaska.edu/SLC/SA/S1A_WV_SLC__1SSV_20190201T001401_20190201T002532_025728_02DC1A_8582.zip']]


```
To give the slightly broader picture, I found that `earthaccess` makes it very simple to temporary S3 credentials for ASF. I then wanted to look up the S3 bucket for Sentinel-1 data to get direct access from an EC2 instance.  I see that the results from `search_data` only have the HTTPS url... so I'm not sure if there's other parameters I need to set, or if there's another existing way to do it.

**Reply by betolink** [reply, marked as answer]
@scottstanie no worries, the documentation needs a lot of work. For this use case you could do something like...  

```python
import earthaccess

# this will use environment variables, or a netrc if not found it will ask us for our EDL credentials
auth = earthaccess.login()
s3_credentials = earthaccess.get_s3_credentials("ASF")

results = earthaccess.search_data(short_name="SENTINEL-1A_SLC",
                                  granule_name="S1A_WV_SLC__1SSV_20190201T001401_20190201T002532_025728_02DC1A_8582")
# technically we will get only one result back 
s3_links = [g.data_links(access="direct")[0] for g in results]
s3_links

```

And then you could use the S3 credentials with other libraries we want to use to access that file. Just remember they last for 1 hour only.
We could also use `earthaccess` to download the file and not worry about the credentials with:

```python
import earthaccess

# this will use environment variables, or a netrc if not found it will ask us for our EDL credentials
auth = earthaccess.login()

results = earthaccess.search_data(short_name="SENTINEL-1A_SLC",
                                  granule_name="S1A_WV_SLC__1SSV_20190201T001401_20190201T002532_025728_02DC1A_8582")

files = earthaccess.download(results, "my_slc_data")

```

Looking forward to know if this helps!

**Comment by jhkennedy** [comment]
@betolink `asf_search`, for example, has a fallback `.umm` method for granules that gives you the entire UMM JSON record, which would give you access to *everything* in CMR. Would it be useful to also do that here?

**Reply by jhkennedy** [reply]
Oh nevermind, I missread; this is for passing through for searching, not reporting resuts.

**Reply by betolink** [reply]
yeah this is for the search, on the results side we do have the whole UMM record with

```python
results = earthaccess.search_data(params)
# first granule from the results
granule = results[0]
# we have meta and umm with the full UMM record
granule["umm"]
```
